parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -683,4 +723,77 @@ removeKubeletFlag() {
     fi
 }
 
+oras_login_with_kubelet_identity() {
+    local acr_url=$1
+    local client_id=$2
+    local tenant_id=$3
+
+    if [ -z "$client_id" ] || [ -z "$tenant_id" ]; then
+        echo "client_id or tenant_id are not set. Oras login is not possible, proceeding with anonymous pull"
+        return 
+    fi
+
+    test_image="/mcr/hello-world:latest"
+    retrycmd_can_oras_discover_acr 10 5 $acr_url$test_image
+    ret_code=$? 
+    if [[ $ret_code -eq 0 ]]; then
+        echo "anonymous pull is allowed for acr '$acr_url', proceeding with anonymous pull"
+        return
+    elif [[ $ret_code -ne 1 ]]; then
+        echo "failed with an error other than unauthorized, exiting.."
+        return $ret_code
+    fi
+
+    set +x 
+    access_url="http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https://management.azure.com/&client_id=$client_id"
+    raw_access_token=$(retrycmd_get_access_token_for_oras 10 5 $access_url)
+    ret_code=$? 
+    if [ $ret_code -ne 0 ]; then
+        echo "failed to retrieve access token: $ret_code"
+        return $ret_code
+    fi
+    if [ -z "$raw_access_token" ] || [[ "$raw_access_token" == *"error"* ]]; then
```

Comment:
don't think we need to check if $raw_access_token is empty again since if it's empty ret_code will be non-zero, unless I'm missing something